modules/utxo-lib/src/bitgo/transaction.ts

```diff
@@ -15,9 +15,20 @@ import { ZcashNetwork, ZcashTransaction } from './zcash/ZcashTransaction';
 export function createTransactionFromBuffer<TNumber extends number | bigint = number>(
   buf: Buffer,
   network: Network,
-  { version }: { version?: number } = {},
-  amountType: 'number' | 'bigint' = 'number'
+  { version, amountType }: { version?: number; amountType?: 'number' | 'bigint' } = {},
+  deprecatedAmountType?: 'number' | 'bigint'
 ): UtxoTransaction<TNumber> {
+  if (amountType) {
+    if (deprecatedAmountType && amountType !== deprecatedAmountType) {
+      throw new Error(`invalid arguments`);
+    }
+  } else {
+    if (deprecatedAmountType) {
+      amountType = deprecatedAmountType;
+    } else {
+      amountType = 'number';
+    }
+  }
```

Comment:
NIT,  can avoid some nesting if else
```suggestion
  if (amountType && deprecatedAmountType && amountType !== deprecatedAmountType) {
      throw new Error(`invalid arguments`);
  } else if (deprecatedAmountType)  {
      amountType = deprecatedAmountType;
    } else {
      amountType = 'number';
  }
```

Reply:
This is not equivalent - if only `amountType` is set it would be overwritten in the `else` branch. 

There are four cases to deal with (two cases for both arguments). Every case is different. The original code makes it easier to reason about.

Reply:
Ohh, yeah. Got it!!